model/dependencies.go

```diff
@@ -21,9 +21,8 @@ const (
 )
 
 // ApplyDefaults applies defaults to the DependencyLink.
-func (d DependencyLink) ApplyDefaults() DependencyLink {
+func (d *DependencyLink) ApplyDefaults() {
```

Comment:
No need to change return type

Reply:
In this case we are taking a pass by value  modifying it and then returning it again. Isn't it better to use a pointer receiver here and remove the return type ?